packages/pigeon/lib/generator_tools.dart

```diff
@@ -219,3 +219,33 @@ void addLines(Indent indent, Iterable<String> lines, {String? linePrefix}) {
     indent.writeln('$prefix$line');
   }
 }
+
+/// Recursively merges [modification] into [source].  In other words, whenever
```

Comment:
FWIW I find "source" very confusing here; I generally think of the thing being merged into as the target, and source suggests the opposite. I.e., I would find "source" and "target" to be reasonable names for what you've called "modification" and "source" respectively, so I think avoiding the word "source" altogether is probably the safest option. ("base", "target", or "original" would all be reasonable names for it from my perspective.)

Reply:
sgtm, "base" it is